app/controllers/saved_items_controller.rb

```diff
@@ -0,0 +1,19 @@
+class SavedItemsController < ApplicationController
+  before_action :set_listing, only: [:create, :destroy]
+
+  def create
+    current_user.saved_items.create(listing: @listing)
+    redirect_to request.referer, notice: "Listing saved successfully."
```

Comment:
[redirect_back](https://apidock.com/rails/v5.0.0.1/ActionController/Redirecting/redirect_back) does what you want here!